R/config-log.R

```diff
@@ -86,29 +100,104 @@ config_log_impl <- function(.mods) {
     json_files <- json_files[!missing]
   }
 
-  # map over json files and parse relevant fields
-  df <- map_df(json_files, function(.path) {
-
-    .conf_list <- fromJSON(.path)
-
-    if (!all(CONFIG_KEEPERS %in% names(.conf_list))) {
-      warning(paste(
-        glue("{.path} is missing the required keys: `{paste(CONFIG_KEEPERS[!(CONFIG_KEEPERS %in% names(.conf_list))], collapse=', ')}` and will be skipped."),
-        glue("This is likely because it was run with an old version of babylon. Model was run on version {.conf_list$bbi_version}"),
-        "User can call `bbi_current_release()` to see the most recent release version, and call `use_bbi(options('rbabylon.bbi_exe_path'))` to upgrade to the version.",
-        sep = "\n"
-      ))
-      return(NULL)
-    }
+  json_files %>%
+    purrr::map_dfr(parse_bbi_config) %>%
+    dplyr::select(.data[[ABS_MOD_PATH]], everything()) %>%
+    create_config_log_object()
+}
 
-    .conf_list <- .conf_list[CONFIG_KEEPERS]
-    .conf_list[[ABS_MOD_PATH]] <- dirname(.path)
+#' Parse a bbi config file
+#'
+#' @param path A string giving the path to `bbi_config.json`.
+#' @param fields A character vector of fields to include.
+#' @param model_path_field A string giving the field name for the model path.
+#'
+#' @return A list whose elements include
+#'
+#'   * the path to the model file (minus extension)
+#'
+#'   * `fields`
+#'
+#'   * whether the model file has changed
+#'
+#'   * whether the data file has changed
+#'
+#'   The return value is `NULL` if any element of `fields` is not found in
+#'   `path`.
+#'
+#' @keywords internal
+parse_bbi_config <- function(path,
+                             fields = CONFIG_KEEPERS,
+                             model_path_field = ABS_MOD_PATH) {
+  checkmate::assert_file_exists(path)
+  checkmate::assert_character(fields)
+  checkmate::assert_string(model_path_field)
+
+  config <- jsonlite::fromJSON(path)
+
+  if (!all(fields %in% names(config))) {
+    msg <- glue::glue(
+      glue::glue(
+        "{path} is missing the required keys:",
+        "`{paste(fields[!(fields %in% names(config))], collapse = ', ')}`",
+        "and will be skipped.",
+        .sep = " "
+      ),
+      glue::glue(
+        "This is likely because it was run with an old version of babylon.",
+        "Model was run on version {config[['bbi_version']]}",
+        .sep = " "
+      ),
+      glue::glue(
```

Comment:
does this need to be glue? I don't see any variables in the string. I think could just be `paste`

Reply:
I agree it could be `paste()`. I used `glue()` for consistency (the previous two "chunks" do use it).